### PR TITLE
Add support for http basic auth, closes #36

### LINKED ISSUE #36: Auth callbacks
For WebSocket servers that require user/password, the SRWebSocket class should expose a callback for applying the (proper) credentials.

### PR DESCRIPTION
**Pull request by mk**

### CODE REVIEW

**Comment by martinmroz** [comment]
Hey, this seems like a great change and we'd love to take it. Would you mind addressing feedback and then signing our CLA (it's in the README)? Thanks!

File: SocketRocket/SRWebSocket.m [hunk 1]
```diff
@@ -255,6 +255,7 @@ @implementation SRWebSocket {
     NSString *_closeReason;
     
     NSString *_secKey;
+    NSString *_authorization;
```
Comment (martinmroz) [inline SocketRocket/SRWebSocket.m]: If you're feeling super helpful, a better name would be something like `_basicAuthorizationString`.

File: SocketRocket/SRWebSocket.m [hunk 2]
```diff
@@ -527,6 +528,12 @@ - (void)didConnect
     _secKey = [keyBytes SR_stringByBase64Encoding];
     assert([_secKey length] == 24);
     
+    // set header for http basic auth
+    if(_url.user && _url.password) {
```
Comment (martinmroz) [inline SocketRocket/SRWebSocket.m]: Further, to be super-defensive, should be `if (_url.user.length && _url.password.length)` to protect against empty strings.

**Comment by mk** [comment]
@martinmroz I made the suggested changes, merged master and signed the CLA.

**Comment by dfed** [comment]
@mk thanks for updating! looks great. Merging :tada:

**Pull request merged**